### PR TITLE
Abhi/dylib speedup tests

### PR DESCRIPTION
**Pull request by pretentious7**
This creates a sub crate for tests, and has that crate use a plane 2 dylib. The only thing that really changes for users is now you've to run `cargo test --all` to run the integration tests.

recompiling tests after adding a println or something now takes 10 seconds! which is still a bit long, but it's a lot better than crashing my computer.

(as for why the linking was so slow, the debug libplane rlib is 134 mb, the linkers operations on this  in both space and time are apparently ~O(N), for the tests we did that for each test, and since it had to be relinked with every tiny change...)

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -2,5 +2,7 @@
 resolver = "2"
 members = [
     "plane",
-    "plane/tests/plane-test-macro",
+	"plane/plane-dynamic",
```
Comment (paulgb) [inline Cargo.toml:5]: nit: spaces not tabs

File: plane/Cargo.toml [hunk 2]
```diff
@@ -2,6 +2,7 @@
 name = "plane"
 version = "0.2.0"
 edition = "2021"
+autotests = false
```
Comment (paulgb) [inline plane/Cargo.toml:5]: why was this necessary?

**Review by paulgb — APPROVED** [review]
this is great! Left a couple comments, otherwise should be good to go.

File: plane/Cargo.toml (hunk 2 above)
Comment (pretentious7) [reply plane/Cargo.toml:5]: otherwise it looks for tests in /tests, but actually since we aren't providing another target...

Comment (pretentious7) [reply plane/Cargo.toml:5]: ah that's right it's cause I was calling plane-tests tests earlier and it would complain

Comment (pretentious7) [reply plane/Cargo.toml:5]: thanks

Comment (paulgb) [reply plane/Cargo.toml:5]: but we no longer have a `/tests` directory at that level, right?

**Review by paulgb — APPROVED** [review]
